src/db_process.c

```diff
@@ -55,6 +56,52 @@ char *get_dbname() {
 	return dbname;
 }
 
+/*
+ * db_table_exists(db, tname)
+ * Check if the table with the provided table name tname exists in database db.
+ * Returns 0 if the table doesn't exist, as no data can be retrieve for the
+ * request, this usually result in program exit.
+ * Returns 1 if the table exists.
+ */
+int db_table_exists(sqlite3 *db, char *tname)
+{
+	int exists = 0;
+
+	int rc;
+	sqlite3_stmt *stmt;
+	const char *check_table_q = "SELECT 1 FROM sqlite_master where type='table' and name=?";
+	
+	rc = sqlite3_prepare_v2(db, check_table_q, -1, &stmt, NULL);
+
+	if (rc != SQLITE_OK) {
+		errx(1, "SQL error: %s\n", sqlite3_errmsg(db));
+	}
+
+	sqlite3_bind_text(stmt, 1, tname, -1, SQLITE_TRANSIENT);
+	rc = sqlite3_step(stmt);
+	
```

Comment:
```suggestion
	sqlite3_finalize(stmt);
```
instead of duplicating it later.